packages/gcc10.rb

```diff
@@ -3,194 +3,12 @@
 class Gcc10 < Package
   description 'The GNU Compiler Collection includes front ends for C, C++, Objective-C, Fortran, Ada, and Go.'
   homepage 'https://www.gnu.org/software/gcc/'
-  version '10.2.1-0f64'
+  version '10.3.0'
   license 'GPL-3'
   compatibility 'all'
-  source_url 'https://github.com/gcc-mirror/gcc/archive/0f64123bde80a37c8d9aced69405e71848a23b95.zip'
-  source_sha256 '87f1b3344db6cdcb7295e63c031e00f2c7755c2656cf9603aeb32cf5a5539ec7'
+  
+    is_fake
 
-  binary_url ({
-     aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gcc10/10.2.1-0f64_armv7l/gcc10-10.2.1-0f64-chromeos-armv7l.tar.xz',
-      armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gcc10/10.2.1-0f64_armv7l/gcc10-10.2.1-0f64-chromeos-armv7l.tar.xz',
-        i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gcc10/10.2.1-0f64_i686/gcc10-10.2.1-0f64-chromeos-i686.tar.xz',
-      x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gcc10/10.2.1-0f64_x86_64/gcc10-10.2.1-0f64-chromeos-x86_64.tar.xz',
-  })
-  binary_sha256 ({
-     aarch64: 'dc0760c6fc123a48a72a6f5492f56d2be4273c29be3dfce6e7b202521c659365',
-      armv7l: 'dc0760c6fc123a48a72a6f5492f56d2be4273c29be3dfce6e7b202521c659365',
-        i686: '52850f19acb5c7def3b7116a5873a5c68c4f9a7fa254b5198deacff36f1e15aa',
-      x86_64: '438ae78481cbbfcc38c6d917498f6b6f0681627a798b263bb9b8ef222ee38b35',
-  })
+    depends_on 'libssp'
```

Comment:
I guess I don't quite understand why we need a fake package for gcc10 if it only depends on libssp.

Reply:
With this upgrading gcc10 should:
- Remove the current gcc10
- Install libssp
So effectively this leads to a big drop in disk space used on most systems, without hopefully causing any compatibility issues for programs which might need libssp from gcc10. (This was the big issue with needing to use crewfix, no?)

Reply:
maybe we can use the libssp from gcc11?

Reply:
@saltedcoffii We are in this PR. :)

Reply:
OK, so I think I understand now why this is needed.  We eventually want to replace gcc10 with gcc11 but this provides an upgrade path.  Here is a wrinkle.  Why not just leave gcc10 with users that haven't upgraded yet?  New installs can have gcc11 but what does it hurt to allow users to stay at gcc10?  I'm more on the side of isolating gcc10, gcc11 and libssp than worrying about an upgrade.  Would it make sense to concentrate on an update-alternatives solution to allow switching between gcc versions?  We could provide a suggestion to upgrade to gcc11 from `crew update` but it would have to swap out gcc10.

Reply:
What I've done is not build gcc11 w/o libssp, but just pull libssp out of the install for gcc11. That seems to be much more stable. I think it could be fine to rebuild gcc10 this way, but we just then need to get gcc10 built that way, and leave a dependency for libssp in gcc 10 10.3.0.

That lets people keep gcc11. Let me see if I can make that happen...

Reply:
(but we need to figure out a way to have buildessential be ok with either `gcc10` or `gcc11`, otherwise any build will make crew install gcc10 or gcc11.

Reply:
Yeah, we want to avoid the overlap of gcc10 and gcc11.  Both cannot be installed at the same time since they share the same executables and probably the same libraries.

Reply:
I still think some sort of provides ___ is the ideal way forward.

If crew is already building a list of compatible packages, it should be able to figure out a list of provides too, add that to the list of packages, and figure out if depends are satisfied...

Reply:
on the contrary. Check out portage's slotting of gcc, it's quite impressive.

https://github.com/gentoo/gentoo/tree/master/sys-devel/gcc/

Reply:
@saltedcoffii those ebuild files are basically skeletons... there's so much abstracted away that I can't make heads or tails of what they're doing there.

(But I don't know whether one can emerge multiple gcc installs at the same time though on gentoo.)

Reply:
The binutils packages might be easier to see slotting then, sorry about that.